src/applications/vaos/components/calendar/CalendarWidget.jsx

```diff
@@ -105,85 +103,27 @@ export default class CalendarWidget extends Component {
     this.setState({ months: updatedMonths });
   };
 
-  isValid = date => {
-    let isValid = true;
-    const { additionalOptions } = this.props.additionalOptions;
-
-    if (additionalOptions) {
-      if (
-        additionalOptions.required &&
-        isDateInSelectedArray(date, this.state.selectedDates)
-      ) {
-        isValid = false;
-        document.getElementById(`date-cell-${date}`).focus();
-        this.setState({ optionsError: additionalOptions.validationMessage });
-      } else {
-        this.setState({ optionsError: null });
-      }
-    }
-
-    return isValid;
-  };
-
-  handleMultiSelect = (date, currentRowIndex) => {
-    const { maxSelections, currentlySelectedDate, onChange } = this.props;
-    let selectedDates = this.props.selectedDates
+  handleMultiSelect = date => {
+    const { onChange } = this.props;
+    const selectedDates = this.props.selectedDates
       ? [...this.props.selectedDates]
       : [];
-    const isInSelectedArray = isDateInSelectedArray(date, selectedDates);
-
-    if (isInSelectedArray) {
-      if (currentlySelectedDate === date) {
-        // If already in array, "unselect" and remove from map
-        selectedDates = removeDateFromSelectedArray(date, selectedDates);
-        this.setState({
-          currentRowIndex: null,
-          optionsError: null,
-        });
-
-        onChange({
-          currentlySelectedDate: null,
-          selectedDates,
-          currentRowIndex,
-        });
-      } else if (
-        !currentlySelectedDate ||
-        (currentlySelectedDate && this.isValid(currentlySelectedDate))
-      ) {
-        this.setState({
-          currentRowIndex,
-        });
-
-        onChange({
-          currentlySelectedDate: date,
-          selectedDates,
-          currentRowIndex,
-        });
-      }
-    } else if (
-      selectedDates.length < maxSelections &&
-      (!currentlySelectedDate ||
-        (currentlySelectedDate && this.isValid(currentlySelectedDate)))
-    ) {
-      if (!this.props.additionalOptions?.required) {
-        selectedDates.push({ date });
-      }
-
-      this.setState({
-        currentlySelectedDate: date,
-        selectedDates,
-        currentRowIndex,
-      });
 
-      onChange({
-        currentlySelectedDate: date,
-        selectedDates,
-        currentRowIndex,
-      });
+    const isInSelectedArray = isDateInSelectedArray(date, selectedDates);
+    // If an option selection is required, don't add this date to selectedDates
+    // array until an option is selected as well
+    if (!isInSelectedArray && !this.props.additionalOptions?.required) {
+      selectedDates.push({ date });
```

Comment:
Nitpicky and not new, but we shouldn't modify this array because it's in Redux and we should treat that data as immutable. You could use `concat` instead.

Reply:
`selectedDates` is declared above using the spread operator.  This creates a new copy of the array, correct? So we wouldn't be mutating the data:

```
const selectedDates = this.props.selectedDates
      ? [...this.props.selectedDates]
      : [];
```

Reply:
Yep, you're right. The weird github diff gap threw me off.